api/category.js

```diff
@@ -1,15 +1,33 @@
 import axios from "axios";
 import chalk from "chalk";
+import PromptSync from "prompt-sync";
+import { finalData } from "../data/finalData.js";
+
+const prompt = PromptSync();
 
 export const listCategories = async () => {
+    let categoryData = {};
     try {
-        const categoryData = await axios.get("https://api.chucknorris.io/jokes/categories");
+        categoryData = await axios.get("https://api.chucknorris.io/jokes/categories");
         categoryData.data.forEach((category, i) => {
             console.log(chalk.green("[" + (i + 1) + "]") + " " + chalk.green(category));
-
         });
     } catch (error) {
         console.error;
     }
     console.log("\n" + chalk.bgGreen("Please enter one of the above categories to hear a joke!"));
+    const chosenCategory = prompt(">");
+    // console.log(Number(chosenCategory));
+    if (isNaN(Number(chosenCategory))) {
+        finalData.category = chosenCategory;
+        return chosenCategory;
+
+    } else {
+        finalData.category = categoryData.data[chosenCategory - 1];
+        return categoryData.data[chosenCategory - 1];
```

Comment:
As you have wrap the api call inside a try and catch , if the code enters the catch block the categoryData will not have data property , thus categoryData.data might not work ? Maybe you can wrap all the above code also in try block .